CMSgov/design-system
# [WNMGDS-2922] Implement ds-modal-dialog component

## Summary

- [Ticket](https://jira.cms.gov/browse/WNMGDS-2922)

## How to test

1. Run `yarn test:unit:wc` and confirm there are no failures
2. Checkout the Storybook stories and confirm the functionality matches the React component
3. Checkout the Astro example and make sure it's reasonable/works in the way you expect.
4. Test with MacOS VoiceOver and confirm that it is read the same way as the React component.

## Checklist
